Alloy/Shaders/ShaderStructures.h

```diff
@@ -0,0 +1,26 @@
+//
+//  ShaderStructures.h
+//  Alloy
+//
+//  Created by Eugene Bokhan on 23/04/2019.
+//
+
+#ifndef ShaderStructures_h
+#define ShaderStructures_h
+
+#include <simd/simd.h>
+
+typedef struct Rectangle {
+    vector_float2 topLeft;
+    vector_float2 bottomLeft;
+    vector_float2 topRight;
+    vector_float2 bottomRight;
+} Rectangle;
+
+typedef struct Line {
```

Comment:
I don't think storing that much is nice idea. I would just introduce PolyLineRenderer that have uniform width across its segments and can draw a series of points